plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -25,48 +25,39 @@ type StatusBadge = {
 	explanation?: string;
 };
 
-function getStatusBadges( subscription: Subscription ): StatusBadge[] {
-	const badges: StatusBadge[] = [];
-
-	if ( subscription.local.installed && ! subscription.active ) {
-		badges.push( {
-			text: __( 'Not connected', 'woocommerce' ),
-			level: StatusLevel.Warning,
-			explanation: __(
-				'To receive updates and support, please connect your subscription to this store.',
-				'woocommerce'
-			),
-		} );
-	}
-
+function getStatusBadge( subscription: Subscription ): StatusBadge | false {
 	if ( subscription.product_key === '' ) {
-		badges.push( {
+		/**
+		 * If there is no subscription, we don't need to check for the expiry.
```

Comment:
Just to check – we don't need to worry about subscriptions for free products that don't have product keys?

Reply:
It should be ok.
Only products with keys or Woo headers will be shown here.
AFAIK all free products that don't support keys, also don't have Woo headers and won't be listed.